File: storage_guard_backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Support\Facades\Log;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Casts\Attribute;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Product extends Model
{
    use HasFactory;
    protected $fillable = [
    'name',
    'description',
    'production_date',
    'expity_date',
    'max_temp',
    'min_temp',
    'max_humidity',
    'min_humidity',
    'safety_status'
    ];

    public function operations()
    {
        return $this->hasManyThrough(Operation::class, ProductsList::class, 'product_id', 'id', 'id', 'operation_id');
    }
}

File: Dashboard/lib/models/product.dart
class Product {
  final int? id;
  final String name;
  final String description;
  final String productionDate;
  final String expiryDate;
  final double maxTemperature;
  final double minTemperature;
  final double maxHumidity;
  final double minHumidity;
  final int? safetyStatus;

  Product({
     this.id,
    required this.name,
    required this.description,
    required this.productionDate,
    required this.expiryDate,
    required this.maxTemperature,
    required this.minTemperature,
    required this.maxHumidity,
    required this.minHumidity,
     this.safetyStatus,
  });

  factory Product.fromJson(Map<dynamic, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      productionDate: json['production_date'],
      expiryDate: json['expiry_date'],
      maxTemperature: json['max_temp'],
      minTemperature: json['min_temp'],
      maxHumidity: json['max_humidity'],
      minHumidity: json['min_humidity'],
      safetyStatus:json['safety_status'],
    );
  }

  Map<String, dynamic> toJson()  => {

      'name': name,
      'description': description,
      'production_date': productionDate,
      'expiry_date': expiryDate,
      'max_temp': maxTemperature,
      'min_temp': minTemperature,
      'max_humidity': maxHumidity,
      'min_humidity': minHumidity,

    };
  }
